Sources/DVR/SessionDataTask.swift

```diff
@@ -111,12 +117,15 @@ final class SessionDataTask: URLSessionDataTask {
                 }
             }
 
+            if filteredRequest == nil {
+                persistInteraction = false
+            }
+            
             if persistInteraction {
-                this.interaction = Interaction(request: filteredRequest, response: filteredResponse, responseData: filteredData)
+                this.interaction = Interaction(request: filteredRequest!, response: filteredResponse, responseData: filteredData)
                 this.session.finishTask(this, interaction: this.interaction!, playback: false)
             } else {
-                this.interaction = Interaction(request: filteredRequest, response: response, responseData: data)
```

Comment:
Why is this line being removed?

Reply:
Because the line following was replaced with an implementation that doesn't require an interaction. When you don't want the interaction, the previous implementation would still create an interaction to act as a shell to pass along the filtered response data. Even though logically it worked, it was confusing to have code that says "don't persist an interaction" and yet there was an interaction created. So I passed the filtered response to a new function that could handle the data without an interaction, which allowed me to remove this line.